solr/core/src/java/org/apache/solr/search/neural/KnnQParser.java

```diff
@@ -72,7 +78,41 @@ public Query parse() {
 
     DenseVectorField denseVectorType = (DenseVectorField) fieldType;
     float[] parsedVectorToSearch = parseVector(vectorToSearch, denseVectorType.getDimension());
-    return denseVectorType.getKnnVectorQuery(schemaField.getName(), parsedVectorToSearch, topK);
+
+    return denseVectorType.getKnnVectorQuery(
+        schemaField.getName(), parsedVectorToSearch, topK, getFilterQuery());
+  }
+
+  private Query getFilterQuery() throws SolrException {
+    if (!isFilter()) {
+      String[] filterQueries = req.getParams().getParams(CommonParams.FQ);
+      if (filterQueries != null && filterQueries.length != 0) {
+        return computeFilterQuery(filterQueries);
+      }
+    }
+    return null;
+  }
+
+  private Query computeFilterQuery(String[] filterQueries) throws SolrException {
```

Comment:
do we really need two private methods:
getFilterQuery()
computeFilterQuery(String[] filterQueries)

Can't we just have one "buildFilterQuery()" ?

They don't appear to be that complex to require a split, but I am happy to reconsider if there was some deep motivation

Reply:
changed